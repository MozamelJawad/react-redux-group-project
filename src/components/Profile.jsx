import React from 'react';
import '../redux/profiles/profile.css';
import { useSelector } from 'react-redux';

function Profile() {
  const { rockets } = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <div className="profile">
      <div className="mission">
        <h2>My Missions</h2>

      </div>

      <div className="rockets">
        <h2>My Rockets</h2>
        { reservedRockets.length > 0 ? (
          <div className="mission-list">
            {
                    reservedRockets.map((rocket) => <p key={rocket.id} className="profile-list">{rocket.name}</p>)
                }
          </div>
        ) : <p className="notExist">There is no reserved rocket.</p>}
      </div>
    </div>
  );
}

export default Profile;
