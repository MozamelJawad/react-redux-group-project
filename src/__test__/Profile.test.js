import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import rocketsReducer from '../redux/rockets/rocketsSlice';
import missionsReducer from '../redux/missions/missionsSlice';
import Profile from '../components/Profile';

describe('Render profile components correctly', () => {
  test('Render joined mission and reserved rockets', () => {
    const store = configureStore({
      reducer: {
        rockets: rocketsReducer,
        missions: missionsReducer,
      },
      initialState: {
        rockets: {
          rockets: [
            { id: 1, name: 'Rocket A', reserved: true },
            { id: 2, name: 'Rocket B', reserved: false },
          ],
        },
        missions: {
          missions: [
            { id: 1, name: 'Missoion A', join: true },
            { id: 2, name: 'Missoion B', join: false },
          ],
        },
      },
    });

    const { container } = render(
      <Provider store={store}>
        <Router>
          <Profile />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
