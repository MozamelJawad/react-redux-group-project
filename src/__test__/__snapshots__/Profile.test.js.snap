// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render profile components correctly Render joined mission and reserved rockets 1`] = `
<div>
  <div
    class="profile"
  >
    <div
      class="mission"
    >
      <h2>
        My Missions
      </h2>
      <p
        class="notExist"
      >
        There is no joined mission.
      </p>
    </div>
    <div
      class="rockets"
    >
      <h2>
        My Rockets
      </h2>
      <p
        class="notExist"
      >
        There is no reserved rocket.
      </p>
    </div>
  </div>
</div>
`;
